usaspending_api/bulk_download/tests/test_download.py

```diff
@@ -159,59 +164,31 @@ def test_download_status_nonexistent_file_404(client):
     assert resp.status_code == status.HTTP_404_NOT_FOUND
 
 
-def sort_function(agency):
-    return agency["toptier_code"]
-
-
-@pytest.mark.skip
 def test_list_agencies(client, award_data):
     """Test transaction list agencies endpoint"""
-    resp = client.post("/api/v2/bulk_download/list_agencies", content_type="application/json", data=json.dumps({}))
-
-    all_toptiers = [
-        {"name": "Bureau of Things", "toptier_code": "100"},
-        {"name": "Bureau of Stuff", "toptier_code": "101"},
-        {"name": "Bureau of Money", "toptier_code": "102"},
-    ]
-
-    index = 0
-    agency_ids = []
-    for toptier in sorted(resp.json()["agencies"]["other_agencies"], key=sort_function):
-        assert toptier["name"] == all_toptiers[index]["name"]
-        assert toptier["toptier_code"] == all_toptiers[index]["toptier_code"]
-        agency_ids.append(toptier["toptier_agency_id"])
-        index += 1
-    assert resp.json()["sub_agencies"] == []
-    assert resp.json()["federal_accounts"] == []
-
     resp = client.post(
         "/api/v2/bulk_download/list_agencies",
         content_type="application/json",
-        data=json.dumps({"agency": agency_ids[0]}),
+        data=json.dumps({"type": "award_agencies"}),
     )
 
-    assert resp.json()["agencies"] == []
-    assert resp.json()["sub_agencies"] == [{"subtier_agency_name": "Bureau of Things", "subtier_agency_id": 1}]
-    assert resp.json()["federal_accounts"] == []
-
-    resp = client.post(
-        "/api/v2/bulk_download/list_agencies",
-        content_type="application/json",
-        data=json.dumps({"agency": agency_ids[1]}),
-    )
-
-    assert resp.json()["agencies"] == []
-    assert resp.json()["sub_agencies"] == [{"subtier_agency_name": "Bureau of Stuff", "subtier_agency_id": 2}]
-    assert resp.json()["federal_accounts"] == []
+    assert resp.status_code == status.HTTP_200_OK
+    assert resp.data == {
+        "agencies": {
+            "cfo_agencies": [],
+            "other_agencies": [
+                {"name": "Bureau of Stuff", "toptier_agency_id": 2, "toptier_code": "101"},
+                {"name": "Bureau of Things", "toptier_agency_id": 1, "toptier_code": "100"},
```

Comment:
what assures us that the agencies come out in this order? I still don't see any logic that sorts here.

Reply:
https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/download/v2/download_list_agencies.py#L59-L65